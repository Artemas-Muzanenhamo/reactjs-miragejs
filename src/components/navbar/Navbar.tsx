import React from "react";

interface Props {
    title: string
}

export class Navbar extends React.Component<Props> {
    render(): React.ReactElement {
        return <div className="uk-navbar-container uk-navbar">
            <div className="uk-navbar-center">
                <a href="" className="uk-navbar-item uk-logo">{this.props.title}</a>
            </div>
        </div>
    }
}
