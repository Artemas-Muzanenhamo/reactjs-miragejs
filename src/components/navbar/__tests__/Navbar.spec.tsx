import renderer from 'react-test-renderer';
import {Navbar} from "../Navbar";
import React from "react";

describe('<Navbar/> component', () => {
    it('renderes <Navbar/> component with title', () => {
        let navbarComponentAsJSON = renderer.create(
            <Navbar title="My Navbar title"/>
        ).toJSON();

        expect(navbarComponentAsJSON).toMatchSnapshot();
    });
});
