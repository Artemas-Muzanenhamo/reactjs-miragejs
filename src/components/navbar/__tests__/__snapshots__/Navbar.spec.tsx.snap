// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> component renderes <Navbar/> component with title 1`] = `
<div
  className="uk-navbar-container uk-navbar"
>
  <div
    className="uk-navbar-center"
  >
    <a
      className="uk-navbar-item uk-logo"
      href="/"
    >
      My Navbar title
    </a>
  </div>
</div>
`;
